rust/common/error.rs

```diff
@@ -20,12 +20,20 @@
  *
  */
 
+use std::convert::Infallible;
+
 #[derive(Debug)]
 pub struct ErrorMessage {
     pub code: usize,
     pub message: String,
 }
 
+impl From<Infallible> for ErrorMessage {
+    fn from(_: Infallible) -> Self {
+        unreachable!()
+    }
+}
+
```

Comment:
Needed for `Result<T, Infallible>` to `Result<T, ErrorMessage>` conversion. `Infallible` can never be instantiated, so mark as `unreachable!()`.
`Result<T, Infallible>` is returned by e.g. autogenerated `TryFrom<T>` when `From<T>` is implemented.